CMakeLists.txt

```diff
@@ -507,6 +509,7 @@ add_custom_target(genIR DEPENDS ${IR_GENERATED_SRCS})
 set_source_files_properties(${IR_GENERATOR} PROPERTIES GENERATED TRUE)
 add_library(ir-generated OBJECT ${IR_GENERATED_SRCS} ${EXTENSION_IR_SOURCES})
 add_dependencies(ir-generated ir genIR)
+target_link_libraries(ir-generated ${P4C_LIB_DEPS})
```

Comment:
Hmm, ir-generated is a special library that should be encapsulated by `ir.h`. There should be no include like `ir/ir-generated.h` anywhere. include-fixer sometimes includes it.